### ISSUE TITLE
Sec-Websocket-Protocol header missing in response when establishing websocket connection

### ISSUE THREAD

**Issue by fei-sie** [containous/traefik#2912]
### Do you want to request a *feature* or report a *bug*?
bug

### What did you do?
I am trying to connect to an mqtt broker backend through websocket port. 

The backend is tested. When I bypass traefik and connect to the backend directly through websocket-mqtt client, it works. The Sec-Websocket-Protocol header is included in the response and connection can be established.

Plain websocket connection (without mqtt) also works through traefik without any problem. Since in this case the client request doesn't include the Sec-Websocket-Protocol header, the response doesn't have to include it anyway.

This seems to be addressed in #1945. But I've tried v1.3.6, v1.4.0, v1.5.2, all show the same behavior.

Request header is as follows:
```
GET ws://xx.xx.xx.xx/mqtt HTTP/1.1
Host: xx.xx.xx.xx
Connection: Upgrade
Pragma: no-cache
Cache-Control: no-cache
Upgrade: websocket
Origin: http://mitsuruog.github.io
Sec-WebSocket-Version: 13
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.167 Safari/537.36
Accept-Encoding: gzip, deflate
Accept-Language: de-DE,de;q=0.9,en-US;q=0.8,en;q=0.7
Sec-WebSocket-Key: p2At3fRCQf5llSwMK8gDIw==
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
Sec-WebSocket-Protocol: mqttv3.1
```

### What did you expect to see?
Connection established, with response headers:
HTTP/1.1 101 Switching Protocols
Upgrade: WebSocket
Connection: Upgrade
Sec-WebSocket-Accept: QOgoa9oaAWMABKAX5KmXeL6Y3cA=
Sec-WebSocket-Protocol: mqttv3.1

### What did you see instead?
Connection is not established. Response headders missing Sec-WebSocket-Protocol:
HTTP/1.1 101 Switching Protocols
Upgrade: websocket
Connection: Upgrade
Sec-WebSocket-Accept: WoS9dPn1xGhh5CRW13jC1uv+zJo=

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.5.2
```
However, I have tried 1.4.0 and 1.3.6 as well. All the same behavior.

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Traefik is started from command line under Ubuntu 16.04.
```toml
################################################################
# Global configuration
################################################################
debug = true

logLevel = "DEBUG"

[entryPoints]
    [entryPoints.http]
    address = ":80"

[traefikLog]
# filePath = "/home/ubuntu/traefik/traefik.log"

[accessLog]
filePath = "/home/ubuntu/traefik/access.log"


################################################################
# Web configuration backend
################################################################

# Enable web configuration backend
[web]
address = ":8080"

################################################################
# Docker configuration backend
################################################################

# Enable Docker configuration backend
[docker]
```

The docker-compose file of the backend is as follows:
```
version: '2'

services:
  mqtt:
    image: toke/mosquitto
    container_name: mqtt
    ports:
      - "1883:1883"
      - "9001:9001"
    labels:
      traefik.backend: mqtt
      traefik.frontend.rule: PathPrefixStrip:/mqtt
      traefik.enable: "true"
      traefik.port: "9001"
      traefik.passHostHeader: "true"
```
### If applicable, please paste the log output in debug mode (`--debug` switch)
I have also observed the following error in traefik log when I tried to send message through the connection:
```
ERRO[2018-02-22T16:48:52Z] vulcand/oxy/forward/websocket: Error when copying from client to backend: websocket: close 1006 (abnormal closure): unexpected EOF 
```
Othewise there is no error or warning related to the http response itself.

**Comment by Juliens** [comment]
Thanks for your interest in Træfik.

Can you provide us with more information about your broker, and your mqtt client.
I have tried with [rabbitmq-mqtt](https://www.rabbitmq.com/web-mqtt.html) and [mqtt client](https://www.npmjs.com/package/mqtt), and everything works for me.

**Comment by feischi** [comment]
The client I am using for this test is http://mitsuruog.github.io/what-mqtt/. The broker is [toke/mosquitto](https://hub.docker.com/r/toke/mosquitto) running on an EC2 instance. The docker-compose file to start the broker is provided in the issue description. Thanks.

**Comment by traefiker** [comment]
Closed by #2975.

**Comment by argiris-mat** [comment]
I would like to re-open this because I face the same error with a different setup.
I want to run jupyter notebooks on kubernetes hosted on aws. I expose traefik through an ELB with ssl as protocol and AWS certificates. 

Services work great except websockets. The client keeps reconnecting with Status Code: 101 Switching Protocols. And the server logs ...

> time="2018-04-04T14:24:39Z" level=error msg="vulcand/oxy/forward/websocket: Error when copying from backend to client: websocket: close 1006 (abnormal closure): unexpected EOF" 

traefik version: 1.5.4
kubernetes: 1.9.3

Is this header related? I read in the documentation that Træfik will pass by default the incoming Host header to the upstream resource.
